src/lib/orion_websocket/ws.cpp

```diff
@@ -53,12 +55,41 @@ struct _orion_websocket
 // Private struct for persistent data
 typedef struct
 {
-  unsigned cid;
+  std::vector<std::string> notify;
   char *message;
   char *request;
   int index;
 }data;
 
+static bool isSubscription
+(
+ const std::vector<std::string>& headName,
+ const std::vector<std::string>& headValue,
+ std::string& subId
+)
+{
+  for (unsigned i = 0; i < headName.size(); ++i)
+  {
+    if (headName[i] != "Location")
+      continue;
+
+    size_t pos = headValue[i].find("subscriptions");
+    if (pos == std::string::npos)
+      return false;
+
+    pos = headValue[i].find_last_of('/');
+    if (pos == std::string::npos)
+      return false;
+
+    char buff[25];
```

Comment:
I'd suggest to use (in globals/limits.h)

#define MAX_LENGTH_SUBID 24

and size the buff variable with MAX_LENGTH_SUBID + 1. Otherwise, the raw 24 and 25 may be a bit obscure.

(const is also ok if you prefer that to #deffine)